frontend-mobile/lib/widgets/no_username.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flutter/material.dart';
+
+import '../screens/settings_screen.dart';
+
+class NoUsername extends StatelessWidget {
+  final VoidCallback onUsernameSet;
+
+  const NoUsername({super.key, required this.onUsernameSet});
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          const Text('Please set a username to participate in calls!',
+              style: TextStyle(fontSize: 32)),
+          TextButton(
+            onPressed: () async {
+              await navigateToSettings(context);
+              onUsernameSet();
```

Comment:
What happens if you leave the settings screen without having set the username? In fact, now, we have a nice application of the bloc pattern that we could go through (having a bloc for the username/settings state that re-renders both screens when the settings screen updates its state)

Reply:
U still see the screen to update the settings, no error :)
<img width="390" alt="image" src="https://user-images.githubusercontent.com/17473929/191358035-7dca3d5e-c296-4239-b71a-2961118b3927.png">